### DISCUSSION TITLE
Underwater stabilisation

### DISCUSSION THREAD

**Discussion by anton-belkin** [gopro/labs#844] [Feature Requests]
The problem: GoPro lacks proper underwater stabilisation all the way up to latest hero12. I am really impressed that in air the picture is rock solid, but no matter what I do, underwater is always somewhat shaky. I know limitations of using a stick instead of a frame underwater, but in air in the very same conditions it’s solid.

The reason: I found the reason to be that water refraction causes FOV of the lens to diminish by some 5% compared to air FOV. That means each angular move of the camera translates into 5% more pixels shift in picture than it would have with in-air FOV. That in turn means that hypersmooth undercompensates shaking by 5%. Big or small every share makes its way into footage. 

The solution: calibrate hypersmooth for underwater. Add a setting “optimise hypersmooth for underwater”. This feature could be delivered as a software update to the latest and maybe previous versions of the camera. 

So many recreational divers would say thank you! My primary use of GoPro is underwater. And I value its small size, the ability with little inconvenience take my impressions with me. I will not buy a frame. Post production stabilisation adds distortion due to inability to account for lenses at all. Tools such as gyroflow would require no-hypersmooth video as an input to correct the problem, but that would worsen video compression. The main thing it is the problem of thousands of clients

**Comment by dnewman-gpsw** [comment]
Lack of correct underwater stabilization is known. It is possible to configure the camera so that Gyroflow could be used, but disabling HS is not the only issue, you also need to disable the rolling shutter compensation (oMBERS=1), but the workflow is painful (I've personally tried.) I will see if there can be a future Labs solution. This is not an easy hack.

**Reply by anton-belkin** [reply]
Thank you for your quick reply. 

It may be harder than I think, but the point is it’s a software change, might be even just parameterisation, and definitely not hardware. There might be more places to account for it, but what water does is just changing the lens profile: FOV down by about 30% (I’ve rechecked my calc) and some non-linearity closer to edges. Everything gyroflow does is done already by GP for hypersmooth. And if any it’s open source. 

Having underwater stabilisation would be awesome. No one on the market has that. And GoPro is a standard for underwater video around the world, and for such a solid brand, unstable underwater footage is a kind of a shame. Easy hit for Chinese action cam challengers. 

For now I am going to try gyroflow path. Thank you for the rolling shutter hint. Any other advice you might have? I think of increasing bitrate compensate for bitrate wasted for shakes and setting lens to wide?

**Comment by anton-belkin** [comment]
Hey, we have done some tests and computations with gyroflow community. https://github.com/gyroflow/gyroflow/issues/398

It turns out, that one would achieve vast improvement of underwater stabilisation my merely multiplying amount of angular compensations by factor of 1.33. Non-linear distortion added by water does not exceed 2.5% at edges of 90 degree FOV. 

Shutter compensation error is not visible underwater due to low light and thus long exposures. 

And hey, underwater footage is already bad, when camera is shaking, so it won’t get only better with this feature. For stable camera it won’t get worse either. 

Sorry for an outsider’s judgement, but that sounds super simple. 

If we want to go further and compensate for non-linear distortion caused by water, it can be done parametrically, assuming GoPro uses distortion approximation model similar to GyroFlow. We have done such parametric adjustments there already and proposed a method to do it for any lens.

**Comment by anton-belkin** [comment]
Hey, it looks like the problem is solved in HERO12 Labs firmware v2.32.70, December 12, 2024 using DIVE labs setting.
THANK YOU!

I will try it out with the next opportunity. I will be so happy to throw away my bulky spherical lens underwater case.

**Reply by shokwaav** [reply]
I think the spherical lenses also lead to better corner sharpness, although I agree they add a lot of bulk.
